content-test/selectors/selectors.test.js

```diff
@@ -278,20 +279,39 @@ describe("selectors", () => {
 
       assert.equal(state.rows[0].backgroundColor, "#fff");
     });
-    it("should append first run data when init is true", () => {
-      const state = selectWeightedHighlights({WeightedHighlights: {rows: [], init: true}});
+    it("should only change the rows property and copy others over", () => {
+      const state = selectWeightedHighlights({WeightedHighlights: {init: true, rows: []}});
 
-      assert.equal(state.rows.length, firstRunData.Highlights.length);
+      assert.equal(state.init, true);
     });
-    it("should not append first run data when init is false", () => {
-      const state = selectWeightedHighlights({WeightedHighlights: {rows: []}});
+  });
+  describe("selectAndDedupe", () => {
+    it("should dedupe items", () => {
+      const result = selectAndDedupe({
+        dedupe: ["http://www.mozilla.org"],
+        sites: ["http://www.mozilla.org"]
+      });
 
-      assert.equal(state.rows.length, 0);
+      assert.equal(result.length, 0);
     });
-    it("should only change the rows property and copy others over", () => {
-      const state = selectWeightedHighlights({WeightedHighlights: {init: true, rows: []}});
+    it("should slice the results based on `max` argument", () => {
+      const result = selectAndDedupe({
+        sites: ["http://www.mozilla.org", "http://www.firefox.com"],
+        dedupe: [],
+        max: 1
+      });
 
-      assert.equal(state.init, true);
+      assert.equal(result.length, 1);
+    });
+    it("should append defaults if result length is under the specified limit", () => {
+      const result = selectAndDedupe({
+        sites: ["http://www.mozilla.org"],
+        dedupe: [],
+        max: 5,
+        defaults: firstRunData.Highlights
+      });
+
+      assert.equal(result.length, 1 + firstRunData.Highlights.length);
```

Comment:
since `selectAndDedupe` should still only return a max of `options.max`, can we test a case where the combined length of `options.sites` and `options.defaults` is greater than the max?